src/tap/cmd.rs

```diff
@@ -98,12 +99,27 @@ async fn run(
         );
     };
 
+    let start_time = Instant::now();
+    let stream_duration = match opts.duration {
+        Some(duration) => Duration::from_millis(duration),
+        None => Duration::MAX,
+    };
```

Comment:
nit: consider this may be simpler:
```suggestion
    let stream_duration = opts.duration.map(Duration::from_millis).unwrap_or(Duration::MAX);
```

Reply:
Yea that would work better, though going with @jszwedko's suggestion to remove this entirely